src/components/syncplay/timeSyncManager.js

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable indent */
+
+/**
+ * Module that manages time syncing with server.
+ * @module components/syncplay/timeSyncManager
+ */
+
+import events from 'events';
+import connectionManager from 'connectionManager';
+
+/**
+ * Time estimation
+ */
+const NumberOfTrackedMeasurements = 8;
+const PollingIntervalGreedy = 1000; // milliseconds
+const PollingIntervalLowProfile = 60000; // milliseconds
+const GreedyPingCount = 3;
+
+/**
+ * Class that stores measurement data.
+ */
+class Measurement {
+    /**
+     * Creates a new measurement.
+     * @param {Date} t0 Client's timestamp of the request transmission
+     * @param {Date} t1 Server's timestamp of the request reception
+     * @param {Date} t2 Server's timestamp of the response transmission
+     * @param {Date} t3 Client's timestamp of the response reception
+     */
+    constructor(t0, t1, t2, t3) {
+        this.t0 = t0.getTime();
+        this.t1 = t1.getTime();
+        this.t2 = t2.getTime();
+        this.t3 = t3.getTime();
+    }
```

Comment:
it's better to have descriptive names instead of relying on docs only, like
```suggestion
    /**
     * Creates a new measurement.
     * @param {Date} requestSent Client's timestamp of the request transmission
     * @param {Date} requestReceived Server's timestamp of the request reception
     * @param {Date} responseSent Server's timestamp of the response transmission
     * @param {Date} responseReceived Client's timestamp of the response reception
     */
    constructor(requestSent, requestReceived, responseSent, responseReceived) {
        this.requestSent = requestSent.getTime();
        this.requestReceived = requestReceived.getTime();
        this.responseSent = responseSent.getTime();
        this.responseReceived = responseReceived.getTime();
    }
```